Core/GDCore/Events/CodeGeneration/EventsCodeGenerator.cpp

```diff
@@ -614,10 +614,12 @@ EventsCodeGenerator::GenerateCallback(
   const gd::String actionsDeclarationsCode =
       GenerateObjectsDeclarationCode(callbackContext);
 
-  const gd::String callbackCode = callbackFunctionName + " = function (" +
-                                  GenerateEventsParameters(callbackContext) +
-                                  ") {\n" + actionsDeclarationsCode +
-                                  actionsCode + "}\n";
+  const gd::String callbackCode =
+      callbackFunctionName + " = function (" +
+      GenerateEventsParameters(callbackContext) + ") {\n" +
+      "gdjs.copyArray(asyncObjectsList.getLocalVariablesContainers(), "
+      "runtimeScene._localVariables);\n" +

```

Comment:
nit: move these 2 lines into a "localVariablesRestoreCode" so that it's easier for the reader to see that there is a "restoration" done here.